Stage2 code assuming the target to be 64bit

I can compile from master for x86_64-linux-musl but when trying for arm with the command below I get the following errors.

./build -j4 arm-linux-musleabi mpcorenovfp


```
[ 99%] Building self-hosted component /home/lance/zig-bootstrap/out/build-zig-arm-linux-musleabi-mpcorenovfp/zig1.o
./src/link/MachO/Object.zig:295:53: error: expected type 'usize', found 'u64'
        var code = try self.allocator.alloc(u8, sect.size);
                                                    ^
./src/link/MachO/Object.zig:295:53: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
        var code = try self.allocator.alloc(u8, sect.size);
                                                    ^
./src/link/MachO/Object.zig:295:44: note: referenced here
        var code = try self.allocator.alloc(u8, sect.size);
                                           ^
./src/link/MachO/Object.zig:216:5: note: referenced here
    try self.parseSections();
    ^
./src/link/MachO/Zld.zig:296:17: note: referenced here
                try object.parse();
                ^
./src/link/MachO/Zld.zig:216:5: note: referenced here
    try self.parseInputFiles(files);
    ^
./src/link/MachO.zig:702:13: note: referenced here
            try zld.link(input_files.items, full_out_path);
            ^
./src/Compilation.zig:1447:5: note: referenced here
    try self.bin_file.flush(self);
    ^
./src/main.zig:2089:5: note: referenced here
    try comp.update();
    ^
./src/main.zig:1930:35: note: referenced here
    updateModule(gpa, comp, hook) catch |err| switch (err) {
                                  ^
./src/stage1.zig:47:43: note: referenced here
        stage2.mainArgs(gpa, arena, args) catch |err| fatal("{s}", .{@errorName(err)});
                                          ^
./src/codegen/c.zig:900:58: error: destination type 'codegen.AnyMCValue' has size 16 but source type 'codegen.c.CValue' has size 8
    inst.codegen.mcv = @bitCast(@import("../codegen.zig").AnyMCValue, result);
                                                         ^
./src/codegen/c.zig:900:71: note: referenced here
    inst.codegen.mcv = @bitCast(@import("../codegen.zig").AnyMCValue, result);
                                                                      ^
./src/codegen/c.zig:643:23: note: referenced here
            .block => try genBlock(o, inst.castTag(.block).?),
                      ^
./src/Sema.zig:1781:63: error: expected type 'usize', found 'u64'
            .data = .{ .name = sema.mod.error_name_list.items[int] },
                                                              ^
./src/Sema.zig:1781:63: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
            .data = .{ .name = sema.mod.error_name_list.items[int] },
                                                              ^
./src/codegen/wasm.zig:898:47: error: destination type 'codegen.AnyMCValue' has size 16 but source type 'codegen.wasm.WValue' has size 8
            .mcv = @bitCast(AnyMCValue, WValue{ .block_idx = self.block_depth }),
                                              ^
./src/codegen/wasm.zig:1011:33: error: destination type 'codegen.wasm.WValue' has size 8 but source type 'codegen.AnyMCValue' has size 16
        const wvalue = @bitCast(WValue, br.block.codegen.mcv);
                                ^
./src/codegen/wasm.zig:1011:57: note: referenced here
        const wvalue = @bitCast(WValue, br.block.codegen.mcv);
                                                        ^
./src/translate_c.zig:4230:52: error: expected type 'usize', found 'u64'
            return Tag.array_type.create(c.arena, .{ .len = size, .elem_type = elem_type });
                                                   ^
./src/translate_c.zig:4230:52: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
            return Tag.array_type.create(c.arena, .{ .len = size, .elem_type = elem_type });
                                                   ^
./src/translate_c.zig:4230:41: note: referenced here
            return Tag.array_type.create(c.arena, .{ .len = size, .elem_type = elem_type });
                                        ^
./src/type.zig:2187:45: error: expected type 'usize', found 'u64'
                return int_val.toUnsignedInt();
                                            ^
./src/type.zig:2187:45: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
                return int_val.toUnsignedInt();
                                            ^
./src/translate_c.zig:2340:63: error: expected type 'usize', found 'u64'
    const filler_node = try Tag.array_filler.create(c.arena, .{
                                                              ^
./src/translate_c.zig:2340:63: note: unsigned 32-bit int cannot represent all possible unsigned 64-bit values
    const filler_node = try Tag.array_filler.create(c.arena, .{
                                                              ^
./src/translate_c.zig:2340:52: note: referenced here
    const filler_node = try Tag.array_filler.create(c.arena, .{
                                                   ^
make[2]: *** [CMakeFiles/zig.dir/build.make:326: zig1.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:188: CMakeFiles/zig.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/src/link/MachO/Archive.zig
+++ b/src/link/MachO/Archive.zig
@@ -58,7 +58,7 @@ const ar_hdr = extern struct {
 
     const NameOrLength = union(enum) {
         Name: []const u8,
-        Length: u64,
+        Length: u32,
     };
     fn nameOrLength(self: ar_hdr) !NameOrLength {
         const value = getValue(&self.ar_name);
@@ -70,14 +70,14 @@ const ar_hdr = extern struct {
         } else {
             // Name follows the header directly and its length is encoded in
             // the name field.
-            const length = try std.fmt.parseInt(u64, value[slash_index + 1 ..], 10);
+            const length = try std.fmt.parseInt(u32, value[slash_index + 1 ..], 10);
             return NameOrLength{ .Length = length };
         }
     }
 
-    fn size(self: ar_hdr) !u64 {
+    fn size(self: ar_hdr) !u32 {
         const value = getValue(&self.ar_size);
-        return std.fmt.parseInt(u64, value, 10);
+        return std.fmt.parseInt(u32, value, 10);
     }
 
     fn getValue(raw: []const u8) []const u8 {

--- a/src/link/MachO/Object.zig
+++ b/src/link/MachO/Object.zig
@@ -284,7 +284,7 @@ pub fn parseSections(self: *Object) !void {
     for (seg.sections.items) |sect| {
         log.debug("parsing section '{s},{s}'", .{ parseName(&sect.segname), parseName(&sect.sectname) });
         // Read sections' code
-        var code = try self.allocator.alloc(u8, sect.size);
+        var code = try self.allocator.alloc(u8, @intCast(usize, sect.size));
         _ = try self.file.?.preadAll(code, sect.offset);
 
         var section = Section{
@@ -461,7 +461,7 @@ pub fn parseDebugInfo(self: *Object) !void {
 fn readSection(self: Object, allocator: *Allocator, index: u16) ![]u8 {
     const seg = self.load_commands.items[self.segment_cmd_index.?].Segment;
     const sect = seg.sections.items[index];
-    var buffer = try allocator.alloc(u8, sect.size);
+    var buffer = try allocator.alloc(u8, @intCast(usize, sect.size));
     _ = try self.file.?.preadAll(buffer, sect.offset);
     return buffer;
 }

--- a/src/link/MachO/Zld.zig
+++ b/src/link/MachO/Zld.zig
@@ -2214,7 +2214,7 @@ fn flush(self: *Zld) !void {
         const seg = self.load_commands.items[self.data_segment_cmd_index.?].Segment;
         const sect = &seg.sections.items[index];
 
-        var buffer = try self.allocator.alloc(u8, sect.size);
+        var buffer = try self.allocator.alloc(u8, @intCast(usize, sect.size));
         defer self.allocator.free(buffer);
         _ = try self.file.?.preadAll(buffer, sect.offset);
 

--- a/src/link/MachO/reloc.zig
+++ b/src/link/MachO/reloc.zig
@@ -179,12 +179,12 @@ pub fn parse(
 
 pub const RelocIterator = struct {
     buffer: []const macho.relocation_info,
-    index: i64 = -1,
+    index: i32 = -1,
 
     pub fn next(self: *RelocIterator) ?macho.relocation_info {
         self.index += 1;
         if (self.index < self.buffer.len) {
-            const reloc = self.buffer[@intCast(u64, self.index)];
+            const reloc = self.buffer[@intCast(u32, self.index)];
             log.debug("relocation", .{});
             log.debug("    | type = {}", .{reloc.r_type});
             log.debug("    | offset = {}", .{reloc.r_address});
@@ -199,6 +199,6 @@ pub const RelocIterator = struct {
 
     pub fn peek(self: RelocIterator) macho.relocation_info {
         assert(self.index + 1 < self.buffer.len);
-        return self.buffer[@intCast(u64, self.index + 1)];
+        return self.buffer[@intCast(u32, self.index + 1)];
     }
 };

